app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,72 @@
+class FoodsController < ApplicationController
+  before_action :set_food, only: %i[show edit update destroy]
+  before_action :authenticate_user!
+
+  # GET /foods or /foods.json
+  def index
+    # select all foods that belongs to the current user
+    @foods = current_user.foods
+  end
+
+  # GET /foods/1 or /foods/1.json
+  def show; end
+
+  # GET /foods/new
+  def new
+    @food = Food.new
+  end
+
+  # GET /foods/1/edit
+  def edit; end
+
+  # POST /foods or /foods.json
+  def create
+    @food = Food.new(food_params)
+    @food.user = current_user # Assign the current user to the food's user_id
+
+    respond_to do |format|
+      if @food.save
+        format.html { redirect_to food_url(@food), notice: 'Food was successfully created.' }
+        format.json { render :show, status: :created, location: @food }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @food.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /foods/1 or /foods/1.json
+  def update
+    respond_to do |format|
+      if @food.update(food_params)
+        format.html { redirect_to food_url(@food), notice: 'Food was successfully updated.' }
+        format.json { render :show, status: :ok, location: @food }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @food.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /foods/1 or /foods/1.json
+  def destroy
+    @food.destroy
+
+    respond_to do |format|
+      format.html { redirect_to foods_url, notice: 'Food was successfully deleted' }
+      format.json { head :no_content }
+    end
+  end
```

Comment:
- [ ] When food is added to the shopping list and I try to delete that food, I have an error which is not user friendly, kindly fix this
![Screenshot from 2023-09-15 22-08-21](https://github.com/gdumani/recipify/assets/66097371/850474e6-9106-448f-b88e-90c55a45628b)

Reply:
Done! Thank you